src/test/java/teammates/test/cases/search/InstructorSearchTest.java

```diff
@@ -0,0 +1,133 @@
+
+        InstructorAttributes ins1InCourse1 = dataBundle.instructors.get("instructor1OfCourse1");
+        InstructorAttributes ins2InCourse1 = dataBundle.instructors.get("instructor2OfCourse1");
+        InstructorAttributes ins1InCourse2 = dataBundle.instructors.get("instructor1OfCourse2");
+        InstructorAttributes ins2InCourse2 = dataBundle.instructors.get("instructor2OfCourse2");
+        InstructorAttributes ins3InCourse2 = dataBundle.instructors.get("instructor3OfCourse2");
+        InstructorAttributes insInArchivedCourse = dataBundle.instructors.get("instructorOfArchivedCourse");
+        InstructorAttributes insInUnregCourse = dataBundle.instructors.get("instructor5");
+
+        ______TS("success: search for instructors in whole system; query string does not match anyone");
+
+        InstructorSearchResultBundle results =
+                instructorsDb.searchInstructorsInWholeSystem("non-existent");
+        verifySearchResults(results);
+
+        ______TS("success: search for instructors in whole system; empty query string does not match anyone");
+        
+        results = instructorsDb.searchInstructorsInWholeSystem("");
+        verifySearchResults(results);
+        
+        ______TS("success: search for instructors in whole system; query string matches some instructors");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("instructor1");
+        verifySearchResults(results, ins1InCourse1, ins1InCourse2);
+
+        ______TS("success: search for instructors in whole system; query string should be case-insensitive");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("InStRuCtOr2");
+        verifySearchResults(results, ins2InCourse1, ins2InCourse2);
+
+        ______TS("success: search for instructors in whole system; instructors in archived courses should be included");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("archived");
+        verifySearchResults(results, insInArchivedCourse);
+
+        ______TS("success: search for instructors in whole system; instructors in unregistered course should be included");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("instructor5");
+        verifySearchResults(results, insInUnregCourse);
+
+        ______TS("success: search for instructors in whole system; instructors should be searchable by course id");
+        
+        results = instructorsDb.searchInstructorsInWholeSystem("idOfUnregisteredCourse");
+        verifySearchResults(results, insInUnregCourse);
+        
+        ______TS("success: search for instructors in whole system; instructors should be searchable by course name");
+        
+        results = instructorsDb.searchInstructorsInWholeSystem("idOfTypicalCourse2");
+        verifySearchResults(results, ins1InCourse2, ins2InCourse2, ins3InCourse2);
+        
+        ______TS("success: search for instructors in whole system; instructors should be searchable by their name");
```

Comment:
Don't think its searchable, since displayed name isn't one of the fields included when building the document.